### PR TITLE
Leaderboard Created

### LINKED ISSUE #4: Add pod stats
Add a way to display the number of contributions between fellows. Make a little leaderboard or similar to motivate people to add new issues!

### PR DESCRIPTION
**Pull request by EkjotKaur**
## Please, go through these steps before you submit a PR.

fixed #4 

- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

## What the Pull Request is adding.
 - A leaderboard which shows PRs merged by each fellow
 - It sorts according to the number of PRs
 - Responsive
 
Screenshot (191)

Screenshot (190)

Screenshot (189)

@NoahCardoza Please take a look at my PR.

### CODE REVIEW

File: _sass/leaderboard.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.leaderboard {
+  // padding: 40px 30px;
+  max-width: 800px;
+  margin: auto;
+  margin-bottom: 80px;
+  // background-color: #133667 !important;
```
Comment (NoahCardoza) [inline _sass/leaderboard.scss:6]: If you don't need this styling I'd remove it. No need to push comments code. If you are saving it for a future PR, I'd move it to another file and checkout the `git stash` command.

File: _includes/leaderboard.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<!-- <script>
```
Comment (NoahCardoza) [inline _includes/leaderboard.html:1]: Same with this comment.

File: _data/SerialNo.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+- i: 1
```
Comment (NoahCardoza) [inline _data/SerialNo.yml:1]: What is this/how is this file being used?

File: _data/leaderboard.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+- Fellow: Helen Du
```
Comment (NoahCardoza) [inline _data/leaderboard.yml:1]: It would be best to follow the same naming convention as the other files in `/_data` (kebab-case).

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Great start! Take a look at the reviews I made and ask any questions if you are confused by anything.

Good luck 🚀

File: _sass/leaderboard.scss (hunk 1 above)
Comment (EkjotKaur) [reply _sass/leaderboard.scss]: I've removed commented code

File: _data/SerialNo.yml (hunk 3 above)
Comment (EkjotKaur) [reply _data/SerialNo.yml]: I was trying to do something then figured out another way.
How I have deleted this extra file

File: _data/leaderboard.yml (hunk 4 above)
Comment (EkjotKaur) [reply _data/leaderboard.yml]: changes to kebab-case

**Comment by EkjotKaur** [comment]
@NoahCardoza I've made the changes

File: _data/leaderboard.yml (hunk 4 above)
Comment (NoahCardoza) [reply _data/leaderboard.yml]: Excellent!

**Comment by NoahCardoza** [comment]
This isn't a requirement, but have you thought about using JavaScript to access the GitHub API and count up the PR requests merged in by fellows? Then we wouldn't have to update the data file every time a PR is merged.

If this is something you are interested in looking into let me know, otherwise, this is looking good!

**Comment by EkjotKaur** [comment]
> This isn't a requirement, but have you thought about using JavaScript to access the GitHub API and count up the PR requests merged in by fellows? Then we wouldn't have to update the data file every time a PR is merged.
> 
> If this is something you are interested in looking into let me know, otherwise, this is looking good!

This sounds good. I would try it.

**Comment by NoahCardoza** [comment]
It might also be cool to link the fellow's names to GitHub where their PR's will be listed out.

You could use this link: `https://github.com/MLH-Fellowship/pod-4.1.4-portfolio/pulls/<username>`

**Comment by NoahCardoza** [comment]
Any update on this?

**Comment by EkjotKaur** [comment]
Working on it. I'll complete it till tomorrow.

**Comment by EkjotKaur** [comment]
@NoahCardoza I've added the following:

- [ ] Counting the number of PRs using GitHub API
- [ ] Name of the fellow linked to their pull request page on this repository.

File: _includes/leaderboard.html [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    document.getElementById("tabledata").innerHTML= innerDATA;
+
+  }
+  const fetchingDATA = async (url1, url2) => {
+    const response1 = await fetch(url1);
+    const response2 = await fetch(url2);
+    let data1 = await response1.json();
+    let data2 = await response2.json();
+    console.log(data1);
```
Comment (NoahCardoza) [inline _includes/leaderboard.html:49]: Looks like a debug console.log was left. No need to push that into production.

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Other than that stray console.log I'm loving it! I love live data!

Also, can you revert the changes to the Gemlfile.lock? Just reset it to the old file since we don't want to cause merge conflicts with other PRs. The file needs to be updated but that should be handled in a separate issue/PR.

**Comment by EkjotKaur** [comment]
@NoahCardoza I've made the changes.
I've reverted the changes to the Gemlfile.lock, but it shows one line of difference in the code, maybe a new line, but I'm not able to resolve that.

**Review by NoahCardoza — APPROVED** [review]
Looks good!

**Comment by NoahCardoza** [comment]
> one line of difference in the code, maybe a new line

Some IDE's trim the empty lines at the end of files automatically which can be pretty annoying sometimes 😂 

I was able to fix it from GitHub. Great work!

**Pull request merged**